### PR TITLE
Rewrap Filter and FilterList

### PR DESCRIPTION
**Pull request by davisp**
This updates the first APIs I wrapped to use the both the builder pattern as well as tracking their own context references.

### CODE REVIEW

File: tiledb/api/src/filter.rs [hunk 1]
```diff
@@ -1,345 +1,216 @@
...
-            ctx,
-            FilterOption::WEBP_INPUT_FORMAT,
-            &c_format as *const std::ffi::c_uchar as *const std::ffi::c_void,
+            ffi::FilterOption::POSITIVE_DELTA_MAX_WINDOW,
+            &mut c_width as *mut std::ffi::c_uint as *mut std::ffi::c_void,
         )
+        .map(|_| c_width as u32)
     }
 
     pub fn get_webp_input_format(
```
Comment (rroelke) [inline tiledb/api/src/filter.rs:167]: This being in the `Filter` impl for all of the filter types does point to this being one of multiple steps to clean this up.

File: tiledb/api/src/filter.rs [hunk 2]
```diff
@@ -1,345 +1,216 @@
...
         &self,
-        ctx: &Context,
-    ) -> TileDBResult<WebPFilterInputFormat> {
+    ) -> TileDBResult<ffi::WebPFilterInputFormat> {
```
Comment (rroelke) [inline tiledb/api/src/filter.rs:169]: Hm, we probably want to avoid having `ffi::` in the public APIs.

**Review by rroelke — APPROVED** [review]
I left a few notes but those are things we can improve later.  This looks like a great step

File: tiledb/api/src/filter.rs (hunk 1 above)
Comment (davisp) [reply tiledb/api/src/filter.rs:167]: Absolutely. I'm sure there's a better way to handle this, but it felt like reaching for `Boxy<dyn Something>` which is beyond where I want to go at this stage.

The only thing that makes me ok with the status quo is knowing that these particular API's will likely only be used by use for implementing `Debug` type traits when folks are just looking for a description of the state.

File: tiledb/api/src/filter.rs (hunk 2 above)
Comment (davisp) [reply tiledb/api/src/filter.rs:169]: I definitely wasn't fond of that either, but kinda left it as a "We'll clean up the api vs sys symbols at some point". For the interim, I'll go add a public use to just import `WebPFilterInputFormat` into the `api/filter` namespace to remove the `ffi` tokens.

**Pull request merged**